packages/astro/src/core/logger/vite.ts

```diff
@@ -16,6 +16,10 @@ const vitePageReloadMsg = /page reload (.*)( \(.*\))?/;
 const viteHmrUpdateMsg = /hmr update (.*)/;
 // capture "vite v5.0.0 building SSR bundle for production..." and "vite v5.0.0 building for production..." messages
 const viteBuildMsg = /vite.*building.*for production/;
+// capture "\n  Shortcuts" messages
+const viteShortcutTitleMsg = /.*Shortcuts.*/s;
+// capture "press * + enter to ..." messages
+const viteShortcutHelpMsg = /press.*\+ enter.*to.*/s;
```

Comment:
Definitely is a shame that we have to do it this way.

Reply:
Yeah, not the best but worst case is that the logs would appear "off" if these regexes failed to match. We're only hacking it to make it cleaner for us for now.

Reply:
It's definitely a relief that you're not worried about the fragility of this given what you know from the Vite side. But I've also been burned by this exact same bug in the past more than once, and it's really difficult to catch when it happens (different dep versions installed, "wait, was THIS new message meant to be ignored or not?", "is this message even still used anymore?" etc.).

Still not blocking, but what's the best way to start a discussion in Vite for a feature where these logs get explicit codes? `info(msg, code)`, etc.

Reply:
I think frankly it would be tedious to assign specific codes on Vite side so we have a reliable way to catch it. These are plain logs from Vite's point of view, so it can technically log anything. The logs API is also indirectly exposed to Vite/Rollup plugins.

One issue for example with HMR messages is that it could contain the HMR path, and a `code` couldn't easily represent that. We would still need some regex or property to capture it. We also have existing regexes like this currently, so compromises for now 😄 
https://github.com/withastro/astro/blob/9cfa8399752f2db2a0e0001446a17d36d0a8aef8/packages/astro/src/core/errors/dev/vite.ts#L44

For me, I'd like to pass these through without rewriting as much as possible. I have some ideas to make it possible in the future though.